### ISSUE TITLE
Contao Manager copies contao-manager folder to document root

### ISSUE THREAD

**Issue by fritzmg** [contao/contao-manager#383]
I have updated the Contao Manager to 1.1.2 in a Contao 3 installation (the Contao 3 installation is trakked by trakked.io). The Contao 3 installation is located in the following folder:
```
~/www.example.org/public
```
The document root of the domain points to that folder. The `contao-manager.phar.php` is located within that folder. Thus configuration and cache folder of the Contao Manager is located in
```
~/www.example.org/contao-manager
```
However, after updating, the Contao Manager complains
```
 SECURITY ALERT !!! Unprotected config directory detected 
```
I was confused how this can be, since anything outside the document root should not be accessible in anyway. But then I noticed, as soon as I access
```
https://www.example.org/contao-manager.phar.php
```
the Contao Manager appears to copy the contents of
```
~/www.example.org/contao-manager/
```
to
```
~/www.example.org/public
```
making
```
https://www.example.org/contao-manager/users.json
```
available to the public.

**Comment by aschempp** [comment]
That is correct, it will be installed in the same folder as the Phar, except if the parent folder is `web`.
This has been changed in version 1.1 to fix two issues:
 1. if you have multiple Contao 3 installations, the parent folder is most likely shared, so placing the config in the parent would share it across multiple installations.
 2. If you setup a Contao 4 installation by running the Manager in the root, you can later move it to the web folder without issues.

Be aware that the manager only *copies* the files from parent, this is to work around the first problem, because otherwise all subsequent Contao 3 installation would have no configuration (and therefore expose an unconfigured Manager).

The security protection does two things. It adds a `.htaccess` file to the folder. But because you probably use Nginx, that won't apply. That's why it also checks if the folder is publicly available, and warns you about it.

**Comment by fritzmg** [comment]
Actually we use Apache. I'll have to check why the access rules within the `.htaccess` are not applied.

**Comment by fritzmg** [comment]
Hm, it appears the apache server has `mod_authz_core` enabled, but `Require all denied` isn't working. Is it update safe to edit the `.htaccess` accordingly?

**Comment by Toflar** [comment]
I don't think that's a good idea, I'd rather find out why it's not working. Are `.htaccess` files even considered?

**Comment by fritzmg** [comment]
Yes, it works if I only use
```
Order deny,allow
Deny from all
```

**Comment by Toflar** [comment]
Sounds like a wrong configuration then, no? I mean apparently the if condition evaluates to `true` but the directive does not work. That could affect other parts of the system as well...

**Comment by fritzmg** [comment]
Yeah, I'll check with the provider.
